scripts/doc_crosslinks/generate_cross_link_data.py

```diff
@@ -0,0 +1,120 @@
+    result = re.sub(r'([^A-Z]{2,})V([0-9]+)', r'\g<1> V\g<2>', result)
+    result = re.sub(r'(?<=[a-z])(?=[A-Z]([a-zA-Z]|[0-9]))', r' ', result)
+    result = re.sub(r'([A-Z]+)([A-Z][a-z])', r'\g<1> \g<2>', result)
+    result = re.sub(r'([0-9])([a-zA-Z])', r'\g<1> \g<2>', result)
+    result = re.sub(r' +', ' ', result)
+    return result.split(" ");
+
+
+def capitalize(str):
+    if(str is None or len(str) == 0):
+        return str
+    strFirstCaps = str[0].title() + str[1:]
+    return strFirstCaps
+
+
+def lowerCase(str):
+    if(str is None or len(str) == 0):
+        return str
+    return str.lower()
+
+def pascalCase(str):
+    splits = splitOnWordBoundaries(str)
+    modifiedStr = ""
+    for i in range(0, len(splits)) :
+        modifiedStr += capitalize(lowerCase(splits[i]))
+    return modifiedStr
+
+def getClientClassNameFromMetadata(metadataNode):
+    toSanitize = ""
+    if "serviceId" in metadataNode.keys():
+        toSanitize = metadataNode["serviceId"]
+    clientName = pascalCase(toSanitize)
+    clientName =  removeLeading(clientName , "Amazon")
+    clientName =  removeLeading(clientName, "Aws")
+    clientName = removeTrailing(clientName, "Service" )
+    return   clientName + "Client"
+
+
+def removeLeading(str, toRemove) :
+    if(str is None) :
+        return str
+    if(str.startswith(toRemove)):
+        return str.replace(toRemove, "")
+    return str
+
+def removeTrailing(str, toRemove) :
+    if(str is None) :
+        return str
+    if(str.endswith(toRemove)):
+        return str.replace(toRemove, "")
+    return str
+
+def insertDocsMapToRedirect(apiDefinitionsBasePath, apiDefinitionsRelativePath, templatePath, outputPath):
+    generateDocsMap(apiDefinitionsBasePath, apiDefinitionsRelativePath)
+    output = ""
+    with codecs.open(templatePath, 'rb', 'utf-8') as redirect_template:
+        current_template = redirect_template.read();
+        output = current_template.replace("${UID_SERVICE_MAPPING}", json.dumps(sdks, ensure_ascii=False))
+        output = output.replace("${UID_CLIENT_CLASS_MAPPING}", json.dumps(clientClass, ensure_ascii=False))
+    with open(outputPath, 'w') as redirect_output:
+        redirect_output.write(output)
+
+def Main():
+    parser = argparse.ArgumentParser(description="Generates a Cross-link redirect file.")
+    parser.add_argument("--apiDefinitionsBasePath", action="store")
+    parser.add_argument("--apiDefinitionsRelativePath", action="store")
+    parser.add_argument("--templatePath", action="store")
+    parser.add_argument("--outputPath", action="store")
+    
+    args = vars( parser.parse_args() )
+    argMap = {}
+    argMap[ "apiDefinitionsBasePath" ] = args[ "apiDefinitionsBasePath" ] or "./../services/"
+    argMap[ "apiDefinitionsRelativePath" ] = args[ "apiDefinitionsRelativePath" ] or "/src/main/resources/codegen-resources/service-2.json"
+    argMap[ "templatePath" ] = args[ "templatePath" ] or "./"
+    argMap[ "outputPath" ] = args[ "outputPath" ] or "/"
```

Comment:
Are `templatePath` and `outputPath` always supposed to reference files, rather than a directory path?
If so
1. Could there be more natural defaults than `./` or `/`?
2. It'd be nice if the names of these two arguments differed from the previous, to distinguish their type

Reply:
>Are templatePath and outputPath always supposed to reference files, rather than a directory path?
templatePath is Directory Path and outputPath is File Path. I will rename and set defaults accordingly